design-documents/graph-ql/coverage/catalog.graphqls

```diff
@@ -267,7 +267,8 @@ interface CategoryInterface @typeResolver(class: "Magento\\CatalogGraphQl\\Model
 }
 
 type Breadcrumb @doc(description: "Breadcrumb item."){
-    category_id: Int @doc(description: "Category ID.")
+    category_id: Int @deprecated(reason: "Use the `category_interface_uid` field instead")
+    category_interface_uid: ID! @doc(description: "Unique identifier from objects implementing CategoryInterface")
```

Comment:
Why not just `category_uid` to be more consistent with `category_id`?

Reply:
Provided a bit of background context in https://github.com/magento/architecture/pull/424#discussion_r477626040 for why it's named this way.

I'm a bit conflicted about `category_interface_uid` vs `category_id`. On one hand, `category_uid` seems like a cleaner API surface. 

My concern is that `Category` isn't actually a type in GraphQL (we have `CategoryInterface` and `CategoryTree`) so there's a small bit of misdirection here.

One goal I've had for our refactoring of IDs has been to have a shared naming convention, that would allow us to better analyze the GraphQL schema in the future. Right now, there's no way to programmatically find all references to a specific `ID`. You mentioned in a comment that:

> I guess you also checked all possible mutation input types and variables to support the new uid types.

Right now, any tasks like this are manual and have to be done by a human. The goal with this naming convention is to make it possible to have the schema answer these questions in the future

Reply:
Thank you for the extra background info: I understand now the naming, it's indeed a good idea if you can later automatically analyse and process/refactor the schema to find all these foreign identifiers.